walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,130 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.*;
+import java.util.stream.Stream; 
+import java.util.stream.Collectors;
 
 public final class FindMeetingQuery {
+
+  // The end of the day, i.e. 1440 minutes. 
+  private final int LATEST_TIME = TimeRange.WHOLE_DAY.end();
+
+  // Given a collection of Events and a MeetingRequest, finds all times where event could take place.
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Set<String> desiredGuests = new HashSet<>();
+    desiredGuests.addAll(request.getAttendees());
+    int duration = (int) request.getDuration();
+
+    if (duration > LATEST_TIME){
+        return new ArrayList<>();
```

Comment:
block indentation should be only 2 whitespaces.
https://google.github.io/styleguide/javaguide.html#s4.2-block-indentation